packages/firestore/test/unit/local/local_store.test.ts

```diff
@@ -1486,6 +1535,115 @@ function genericLocalStoreTests(
       .finish();
   });
 
+  it('handles saving bundled documents', () => {
+    return expectLocalStore()
+      .after(
+        bundledDocuments([
+          doc('foo/bar', 1, { sum: 1337 }),
+          deletedDoc('foo/bar1', 1)
+        ])
+      )
+      .toReturnChanged(
+        doc('foo/bar', 1, { sum: 1337 }),
+        deletedDoc('foo/bar1', 1)
+      )
+      .toContain(doc('foo/bar', 1, { sum: 1337 }))
+      .toContain(deletedDoc('foo/bar1', 1))
+      .finish();
+  });
+
+  it('handles saving bundled documents with newer existing version', () => {
+    const query = Query.atPath(path('foo'));
+    return expectLocalStore()
+      .afterAllocatingQuery(query)
+      .toReturnTargetId(2)
+      .after(docAddedRemoteEvent(doc('foo/bar', 2, { sum: 1337 }), [2]))
+      .toContain(doc('foo/bar', 2, { sum: 1337 }))
+      .after(
+        bundledDocuments([
+          doc('foo/bar', 1, { sum: 1336 }),
+          deletedDoc('foo/bar1', 1)
+        ])
+      )
+      .toReturnChanged(deletedDoc('foo/bar1', 1))
+      .toContain(doc('foo/bar', 2, { sum: 1337 }))
+      .toContain(deletedDoc('foo/bar1', 1))
+      .finish();
+  });
+
+  it('handles saving bundled documents with older existing version', () => {
+    const query = Query.atPath(path('foo'));
+    return expectLocalStore()
+      .afterAllocatingQuery(query)
+      .toReturnTargetId(2)
+      .after(docAddedRemoteEvent(doc('foo/bar1', 1, { val: 'to-delete' }), [2]))
+      .toContain(doc('foo/bar1', 1, { val: 'to-delete' }))
+      .after(
+        bundledDocuments([
+          doc('foo/bar', 1, { sum: 1336 }),
```

Comment:
Should this be version 2? 

We may also want to skip bundle document loading for documents that we already have cached at that version. The local documents seem safer than replacing them with a bundled document that could be from an arbitrary source.

Reply:
Not there were no `foo/bar` when loading the bundle. I changed the name to make it less confusing.

Also added a test to check when the existing version is the same.